forge-game/src/main/java/forge/game/trigger/TriggerUnlockDoor.java

```diff
@@ -0,0 +1,50 @@
+package forge.game.trigger;
+
+import java.util.Map;
+
+import forge.game.ability.AbilityKey;
+import forge.game.card.Card;
+import forge.game.card.CardState;
+import forge.game.spellability.SpellAbility;
+import forge.util.Localizer;
+
+public class TriggerUnlockDoor extends Trigger {
+
+    public TriggerUnlockDoor(final Map<String, String> params, final Card host, final boolean intrinsic) {
+        super(params, host, intrinsic);
+    }
+
+    @Override
+    public boolean performTest(Map<AbilityKey, Object> runParams) {
+        if (!matchesValidParam("ValidCard", runParams.get(AbilityKey.Card))) {
+            return false;
+        }
+
+        if (!matchesValidParam("ValidPlayer", runParams.get(AbilityKey.Player))) {
+            return false;
+        }
+
+        if (hasParam("ThisDoor")) {
+            CardState state = (CardState) runParams.get(AbilityKey.CardState);
+            if (!getCardState().equals(state)) {
```

Comment:
@Hanmac
this logic needs to be fixed somehow
right now it fails for copied spells because then the trigger uses the original CardState

Reply:
What's your opinion? Just compare the CardStateName? (after ValidCard compared against `Card.Self` ?)

Reply:
seems safer 🤔

Reply:
is this good?

Reply:
works now